core/web/transfer_controller.go

```diff
@@ -40,9 +41,63 @@ func (tc *TransfersController) Create(c *gin.Context) {
 		return
 	}
 
-	db := tc.App.GetSqlxDB()
-	q := pg.NewQ(db, tc.App.GetLogger(), tc.App.GetConfig())
-	etx, err := bulletprooftxmanager.SendEther(q, chain.ID(), tr.FromAddress, tr.DestinationAddress, tr.Amount, chain.Config().EvmGasLimitTransfer())
+	if tr.FromAddress == utils.ZeroAddress {
+		jsonAPIError(c, http.StatusUnprocessableEntity, fmt.Errorf("withdrawal source address is missing: %v", tr.FromAddress))
+		return
+	}
+
+	if !tr.AllowHigherAmounts {
+		var balance *big.Int
+
+		balanceMonitor := chain.BalanceMonitor()
+
+		if balanceMonitor != nil {
+			balance = balanceMonitor.GetEthBalance(tr.FromAddress).ToInt()
+		} else {
+			balance, err = chain.Client().BalanceAt(c, tr.FromAddress, nil)
+			if err != nil {
+				jsonAPIError(c, http.StatusInternalServerError, err)
+				return
+			}
+		}
+
+		zero := big.NewInt(0)
+
+		if balance == nil || balance.Cmp(zero) == 0 {
+			jsonAPIError(c, http.StatusUnprocessableEntity, fmt.Errorf("balance is too low for this transaction to be executed: %v", balance))
+			return
+		}
+
+		if chain.Config().EvmEIP1559DynamicFees() {
```

Comment:
This check is not necessary. It is safe to always send legacy transactions even if EIP1559 is enabled.

Reply:
Resolved